src/NaCl.Core/Base/SalsaIntrinsics/Salsa64.cs

```diff
@@ -0,0 +1,224 @@
+﻿#if INTRINSICS
```

Comment:
I wonder if we do need the preprocessor directive since we are checking if the hardware supports it